### PR TITLE
Align linter rules with implicit style

### PR DESCRIPTION
**Pull request by razor-x**
This is not some random attempt to bikeshead. The rule updates here are things we already agreed on but were missing rules for.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "root": true,
   "ignorePatterns": ["!.storybook"],
   "rules": {
-    "no-console": "warn"
+    "no-console": "error"
```
Comment (razor-x) [inline .eslintrc.json:5]: I am not sure why this would be a warning over an error.

File: .eslintrc.json [hunk 2]
```diff
@@ -38,14 +38,15 @@
...
       "rules": {
+        "@typescript-eslint/no-import-type-side-effects": "error",
         "@typescript-eslint/consistent-type-imports": [
           "error",
           {
             "fixStyle": "inline-type-imports"
           }
         ],
         "@typescript-eslint/no-unused-vars": "off",
-        "@typescript-eslint/explicit-function-return-type": "off",
```
Comment (razor-x) [inline .eslintrc.json:48]: Closes https://github.com/seamapi/react/issues/67

File: .eslintrc.json [hunk 3]
```diff
@@ -38,14 +38,15 @@
...
+        "react/function-component-definition": "error",
```
Comment (razor-x) [inline .eslintrc.json:49]: This enforces the convention @mikewuu suggested we adopt :)

File: .eslintrc.json [hunk 4]
```diff
@@ -38,14 +38,15 @@
         }
       ],
       "rules": {
+        "@typescript-eslint/no-import-type-side-effects": "error",
```
Comment (razor-x) [inline .eslintrc.json:41]: Finally the missing rule that puts `type` on the outside!

**Review by dawnho — APPROVED** [review]

**Review by xplato — APPROVED** [review]

**Pull request merged**